eo-maven-plugin/src/main/java/org/eolang/maven/Footprint.java

```diff
@@ -0,0 +1,158 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven;
+
+import com.jcabi.log.Logger;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.function.Supplier;
+import org.cactoos.text.IoCheckedText;
+import org.cactoos.text.TextOf;
+
+/**
+ * Program footprint in EO compilation process.
+ * The footprint consists of file in the {@link #main} folder and optionally cached
+ * file in {@link #cache} folder.
+ * Caching is applied if {@link #ver} represents true version in terms of
+ * {@link #versioned(String)} method.
+ * <br/>Usage example:
+ * <code>
+ *  <pre>
+ *    final Footprint footprint = new Footprint(
+ *      version,
+ *      targetRoot,
+ *      cacheRoot
+ *    ).save(program, ext);
+ *  </pre>
+ * </code>
+ * @since 1.0
+ */
+public class Footprint {
+    /**
+     * Path to target root.
+     */
+    private final Path main;
+
+    /**
+     * Version tag.
+     */
+    private final String ver;
+
+    /**
+     * Path to cache root.
+     */
+    private final Path cache;
+
+    /**
+     * Ctor.
+     * @param ver Version tag
+     * @param main Main root
+     * @param cache Cache root
+     */
+    public Footprint(final String ver, final Path main, final Path cache) {
+        this.ver = ver;
+        this.main = main;
+        this.cache = cache;
+    }
+
+    /**
+     * Get content of the program.
+     * @param program Program name
+     * @param ext File extension
+     * @return Content of a file
+     * @throws IOException In case of IO issue.
+     */
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    public String content(final String program, final String ext) throws IOException {
+        final Path cached = new Place(program).make(this.cache.resolve(this.safeVer()), ext);
+        final Path target = new Place(program).make(this.main, ext);
+        final IoCheckedText content;
+        if (Footprint.versioned(this.ver) && cached.toFile().exists()) {
+            content = new IoCheckedText(
+                new TextOf(cached)
+            );
+        } else {
+            content = new IoCheckedText(
+                new TextOf(target)
+            );
+        }
+        return content.asString();
+    }
+
+    /**
+     * Save content.
+     * @param program Program name
+     * @param ext File extension
+     * @param content File content
+     * @throws IOException In case of IO issues
+     */
+    public void save(final String program, final String ext, final Supplier<String> content)
```

Comment:
@mximp what's the point of sending `Supplier` here instead of just `String`? Usually, callbacks are used when we want to delay execution. But here we don't delay anything, we immediately save what's provided.. Or not? If not, than I can't understand the idea of this method. In my understanding, it has to save both to target and to cache, thats it.

Reply:
@yegor256 Method `.get()` of the supplier is called only if the object is not cached. The idea here to allow conditional EO parsing and the decision is made within `Footprint` based on its internals. 
Without this we would have to calculate content (perform the parsing) before passing to `save` method or expose cached/not-cached property to the client to decide whether to parse it or not.